src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,488 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Test.Common;
+using System.Security.Authentication;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Security.Tests
+{
+    public class KDCSetup : IDisposable
+    {
+        private const string Krb5ConfigFile = "/etc/krb5.conf";
+        private const string KDestroyCmd = "kdestroy";
+        private const string SudoCommand = "sudo";
+        private const string ScriptName = "setup-kdc.sh";
+        private const string ScriptUninstallArgs = "--uninstall --yes";
+        private const string ScriptInstallArgs = "--password {0} --yes";
+        private readonly bool _isKrbInstalled ;
+        public readonly string password;
+
+        public KDCSetup()
+        {
+            _isKrbInstalled = File.Exists(Krb5ConfigFile) &&
+                File.ReadAllText(Krb5ConfigFile).Contains(TestConfiguration.Realm);
+            if (!_isKrbInstalled)
+            {
+                password = Guid.NewGuid().ToString("N");
+                int exitCode = RunSetupScript(string.Format(ScriptInstallArgs, password));
+                if (exitCode != 0)
+                {
+                    if (exitCode != 2)
```

Comment:
What does exit code 2 mean, and why is it special?  It'd be good to put it in a named constant that describes what the exit code means.